arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Option.kt

```diff
@@ -679,18 +679,30 @@ public sealed class Option<out A> {
   public inline fun all(predicate: (A) -> Boolean): Boolean =
     fold({ true }, predicate)
 
+  @Deprecated(
+    NicheAPI + "Prefer using the Option DSL or fold",
+    ReplaceWith("fold<Option<Option<B>>>({ None }) { value -> f(value).map(::Some) }")
+  )
   public inline fun <B> crosswalk(f: (A) -> Option<B>): Option<Option<B>> =
     when (this) {
       is None -> this
       is Some -> f(value).map { Some(it) }
     }
 
+  @Deprecated(
+    NicheAPI + "Prefer using the Option DSL or fold",
+    ReplaceWith("fold<Map<K, Option<V>>>({ emptyMap() }) { value -> f(value).mapValues { Some(it.value) } }")
+  )
   public inline fun <K, V> crosswalkMap(f: (A) -> Map<K, V>): Map<K, Option<V>> =
     when (this) {
       is None -> emptyMap()
       is Some -> f(value).mapValues { Some(it.value) }
     }
 
+  @Deprecated(
+    NicheAPI + "Prefer using the Option DSL or fold",
+    ReplaceWith("fold<Option<B>?>({ null }) { value -> f(value)?.let(::Some) }")
```

Comment:
```suggestion
    ReplaceWith(
      "getOrNull()?.let { value -> Some(f(value)) }",
      "arrow.core.Some", "arrow.core.Option"
    )
```

Assumes `getOrNull` is added to `Option<A>` similar to we did for `Either`.

Reply:
`Some(f(value))` wouldn't be correct because `f` returns a nullable value, so we could potentially return `Some(null)` instead of the expected `null`.

Reply:
Right, sorry.